src/direc.rs

```diff
@@ -164,6 +180,18 @@ pub struct DirecVector<T> {
     pub direction: Direction,
 }
 
+/// A scalar multiplication for a vector.
+impl<T> std::ops::Mul<T> for &DirecVector<T>
```

Comment:
Besides implementing `Mul<T> for &DirecVector<T>` , one could also implements `Mul<&T> for &DirecVector<T>`, also `Mul<T> for &DirecVector<T>`  and `Mul<T> for DirecVector<T>`  (in the latter case you don't even need a `Clone`). But this is optional, don't worry, I would implement it myself in case you don't.

Reply:
Am I correct that all of them are indeed wrappers over `Mul<&T> for &DirecVector<T>`?